lcservice-go/service/descriptor.go

```diff
@@ -37,16 +37,25 @@ type ServiceCallback = func(Request) Response
 // should be done at runtime. You may use the
 // helper function `DictToStruct` for this purpose.
 type RequestParamName = string
+type RequestParamType = string
 type RequestParamDef struct {
-	Type        string `json:"type"`
-	Description string `json:"desc"`
-	IsRequired  bool   `json:"is_required"`
+	Type        RequestParamType `json:"type"`
+	Description string           `json:"desc"`
+	IsRequired  bool             `json:"is_required"`
 
 	// Only for "enum" Type
 	Values []string `json:"values"`
 }
 type RequestParams = map[RequestParamName]RequestParamDef
 
+const (
+	RequestParamTypeString = "str"
+	RequestParamTypeEnum   = "enum"
+	RequestParamTypeInt    = "int"
+	RequestParamTypeBool   = "bool"
+	RequestParamTypeFlag   = "flag"
```

Comment:
What's a flag? Is it a bool that is false unless set to true?

Reply:
Yes, something like that.

Reply:
Feels a bit redundant given the presence of `bool`. Or did you have a specific use case in mind?